tests/test_generate_markersd.py

```diff
@@ -0,0 +1,68 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+
+import numpy as np
+from parameterized import parameterized
+
+from monai.apps.pathology.transforms.post.dictionary import GenerateMarkersd
+from monai.utils import min_version, optional_import
+from tests.utils import TEST_NDARRAYS
+
+_, has_skimage = optional_import("skimage", "0.19.3", min_version)
+_, has_scipy = optional_import("scipy", "1.8.1", min_version)
+
+EXCEPTION_TESTS = []
+TESTS = []
+
+np.random.RandomState(123)
+
+for p in TEST_NDARRAYS:
+    EXCEPTION_TESTS.append(
+        [{"keys": "mask", "prob_key": "prob"}, p(np.random.rand(2, 5, 5)), p(np.random.rand(1, 5, 5)), ValueError]
+    )
+
+    EXCEPTION_TESTS.append(
+        [{"keys": "mask", "prob_key": "prob"}, p(np.random.rand(1, 5, 5)), p(np.random.rand(2, 5, 5)), ValueError]
+    )
+
+    EXCEPTION_TESTS.append(
+        [
+            {"keys": "mask", "prob_key": "prob", "markers_key": "old_markers"},
+            p(np.random.rand(1, 5, 5)),
+            p(np.random.rand(1, 5, 5)),
+            KeyError,
+        ]
+    )
+
+for p in TEST_NDARRAYS:
+    TESTS.append(
+        [{"keys": "mask", "prob_key": "prob"}, p(np.random.rand(1, 5, 5)), p(np.random.rand(1, 5, 5)), (1, 5, 5)]
+    )
+
+
+@unittest.skipUnless(has_skimage, "Requires scikit-image library.")
+@unittest.skipUnless(has_scipy, "Requires scipy library.")
+class TestGenerateMarkersd(unittest.TestCase):
+    @parameterized.expand(EXCEPTION_TESTS)
+    def test_value(self, argments, mask, probmap, exception_type):
+        with self.assertRaises(exception_type):
+            GenerateMarkersd(**argments)({"mask": mask, "prob": probmap, "old_markers": 1})
+
+    @parameterized.expand(TESTS)
+    def test_value2(self, argments, mask, probmap, expected_shape):
+        result = GenerateMarkersd(**argments)({"mask": mask, "prob": probmap})
+        self.assertEqual(result["markers"].shape, expected_shape)
```

Comment:
I see that you are comparing only shapes for unittests here and elsewhere. Do you think that it is enough to make sure that the algorithm works fine?
i.e., do these tests fail if I modify the code in an erroneous way?

Reply:
If not possible to have a rigorous code before having the pipeline, I would suggest to create output for each of the components and at least check them visually to make sure that it is not illogical.
@Nic-Ma @wyli, what do you think?

Reply:
Hi @drbeh ,

For this PR, I feel it's not easy to add unit tests to compare the values, instead, I would suggest to verify this PR in the training pipeline later, if got reasonable post-processing output, then let's merge it.
And of course, I totally agree we should add integration test later to cover the whole Pathology pipeline.
What do you think? @KumoLiu @wyli 

Thanks.